consumers/src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -357,7 +361,7 @@ spawnDispatcher ConsumerConfig {..} cs cid semaphore runningJobsInfo runningJobs
 
       pure (batchSize > 0)
 
-    reserveJobs :: Int -> m ([job], Int)
+    reserveJobs :: MonadCatch m => Int -> m ([Either (idx, T.Text) job], Int)
```

Comment:
I don't think you need MonadCatch here, it's a top level `m` with MonadMask constraint.